dds/idl/v8_generator.cpp

```diff
@@ -156,6 +156,191 @@ namespace {
     }
   }
 
+  void gen_copyfrom(const char* tgt, const char* src, AST_Type* type,
+                    const char* prop, bool prop_index = false, bool fun_assign = false, std::ostream& strm = be_global->impl_)
+  {
+    const Classification cls = classify(type);
+    AST_PredefinedType::PredefinedType pt = AST_PredefinedType::PT_void;
+    const std::string propName = prop_index ? std::string(tgt) + "[" + prop + "]" : std::string(tgt) + "." + prop,
+      assign_prefix = fun_assign ? "(" : " = ",
+      assign_suffix = fun_assign ? ")" : "";
+
+    if (cls & CL_SCALAR) {
+      std::string clines;
+
+      if (cls & CL_ENUM) {
+        const std::string underscores =
+          dds_generator::scoped_helper(type->name(), "_"),
+          array = "gen_" + underscores + "_names";
+        clines += "      if (lv->IsNumber()) {\n";
+        clines += "        v8::Local<v8::Integer> li = Nan::To<v8::Integer>(lv).ToLocalChecked();\n";
+        clines += "        " + propName + assign_prefix + "static_cast<" + scoped(type->name()) + ">(li->Value())" + assign_suffix + ";\n";
+        clines += "      }\n";
+        clines += "      if (lv->IsString()) {\n";
+        clines += "        v8::Local<v8::String> ls = Nan::To<v8::String>(lv).ToLocalChecked();\n";
+        clines += "        std::string ss(ls->Length(), ' ');\n";
```

Comment:
Looks like this should use `Utf8Length` to match `WriteUf8`.  (I think this occurs in a few other places too.)

Reply:
the last two examples you're looking at are actually for wide strings, so Length() is correct.